app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt

```diff
@@ -29,22 +29,13 @@ fun FavoriteButton(
             .size(dimensionResource(id = R.dimen.favorite_button_size))
             .clip(CircleShape)
             .padding(dimensionResource(id = R.dimen.small_spacing))
-            .clickable {
-                //!isFavorite
-                onClick(isFavorite)
-                //onClick()={!isFavorite} I get an error saying I can't change val parameters
-            }
+            .clickable(onClick = onClick)
     )
 }
 
-fun onClick(isFavorite: Boolean) {
-    !isFavorite
-}
-
 @Preview
 @Composable
 private fun FavoriteButtonPreview(){
-    FavoriteButton(isFavorite = false, onClick = { /*TODO*/ })
+    var isFavorite by remember { mutableStateOf(false) }
+    FavoriteButton(isFavorite = isFavorite, onClick = { !isFavorite })
```

Comment:
Does this work? `{ !isFavorite }` will return boolean but that boolean is not used. What you need to do is change compose state inside that lambda. It should look like this `onClick = { isFavorite = !isFavorite }`.